pandas/tests/io/excel/test_readers.py

```diff
@@ -148,6 +148,23 @@ def parser(self, *args, **kwargs):
             expected = expected_defaults[read_ext[1:]]
         assert result == expected
 
+    def test_engine_kwargs(self, read_ext):
+        expected_defaults = {
+            "xlsx": {"keep_vba": True},
+            "xlsm": {"keep_vba": True},
+            "xlsb": {"debug": True},
+            "xls": {"encoding": "utf-8", "ignore_workbook_corruption": False},
+            "ods": {},
+        }
+
+        df = pd.read_excel(
+            "test1" + read_ext,
+            sheet_name="Sheet1",
+            index_col=0,
+            **expected_defaults[read_ext[1:]],
+        )
+        assert not df.empty
```

Comment:
I don't think this is a sufficient test. Can you instead pass an invalid argument (e.g. `foo`) and check that the expected exception is raised.

Reply:
@rhshadrach when you say "document which function pandas is using for each engine" do you mean that we should be documenting the class associated with each engine? For example, if I want to use pandas.read_excel to read an .xlrd file, I would be using XlrdReader.load_workbook to read in the file?

Reply:
@rhshadrach when you say "document which function pandas is using for each engine" do you mean that we should be documenting the class associated with each engine? For example, if I want to use pandas.read_excel to read an .xlrd file, I would be using XlrdReader.load_workbook to read in the file.

Reply:
@rhshadrach Would you be able to provide any insight into why `TestMySQLAlchemy.test_to_sql_with_negative_npinf` is failing in all of the Ubuntu tests? That test is skipped when I run it locally on my Ubuntu systemusing PyTest.

![image](https://user-images.githubusercontent.com/45905457/228360773-f800c7ab-2edb-45ff-8c94-0af5a7be56c0.png)


Also, the failing Doc Build isn't giving me any insight into the error either which is strange

Reply:
The SQL test failure looks unrelated; the docs build is failing because of a warning - also looks unrelated.

Reply:
> If the SQL test failure and doc build failures are unrelated is there anything left for me to do?

Not in this regard, no. It is okay to ignore unrelated failures.